### PR TITLE
fix(DB/smart_scripts) Fix Quest: The Touch of Zanzil

### PR DESCRIPTION
**Pull request by Kosiak**
## Changes Proposed:
-  Set correct target for quest credit

## Issues Addressed:

- Closes [[Class quest][Rogue] 20 - The touch of Zanzil (Alliance) - Quest doesn't complete #2978](https://github.com/chromiecraft/chromiecraft/issues/2978)

## SOURCE:

https://www.youtube.com/watch?v=ZlT8nE_HwQo

## Tests Performed:

- Accepted quest. Laid down. Gnome finished his walking and gave quest credit. Complete quest.

## How to Test the Changes:

1. Be a Dwarf Rogue Level 20. Follow the quest chain start from [Mathias and the Defias](https://classic.wowhead.com/quest=2360/mathias-and-the-defias) (.q add 2360)
2. Get to quest [The Touch of Zanzil](https://classic.wowhead.com/quest=2608/the-touch-of-zanzil)
3. Start quest. Lay down wait until gnome finished walking and gives u quest credit.
4. Complete quest.

## Known Issues and TODO List:

- According to video there is some delays while he walking around and there is some phrases exist in **creature_text** which he not using. I think quest not scripted to 100%.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nefertumm — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1647391921098550500.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1647391921098550500');
+
+-- Update target for quest credit
+UPDATE `smart_scripts` SET `target_type` = 21, `target_param1` = 5 WHERE `entryorguid` = 7207 AND `id` = 1;
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1647391921098550500.sql:4]: Only 5 yards of max distance? (target_param1)

Comment (Kosiak) [reply data/sql/updates/pending_db_world/rev_1647391921098550500.sql:4]: Quest giver to player is very close at the end of waypoint so it should be enough. I tested from 2 to 5. Found 5 will work better if player will lay abit further from quest giver.

Comment (Nefertumm) [reply data/sql/updates/pending_db_world/rev_1647391921098550500.sql:4]: Hmm but player can move away from the npc. But well, their fault I think :P

**Comment by Nefertumm** [comment]
For further PRs that you do regarding smart_scripts, we usually use DB/SAI as title instead of smart_scripts :)
Thank you for your contribution!

**Pull request merged**